### DISCUSSION TITLE
NullCache isn't initialized properly

### DISCUSSION THREAD

**Discussion by oittaa** [sh4nks/flask-caching#240] [Q&A]
Maybe I've misunderstood something, but if I try to initialize `NullCache` like `SimpleCache` is done in the README.md, something goes wrong and `has` method is missing.

## Test 1
```python
from flask_caching import Cache
cache = Cache(config={"CACHE_TYPE": "NullCache"})
print(cache.has("test"))
```
Output
```
Traceback (most recent call last):
  File "2.py", line 3, in <module>
    print(cache.has("test"))
AttributeError: 'Cache' object has no attribute 'has'
```

## Test 2
If I import `NullCache` directly, `has` method works as expected and always returns `False`.
```python
from flask_caching.backends.nullcache import NullCache
cache = NullCache()
print(cache.has("test"))
```
Output
```
False
```

**Comment by sh4nks** [comment]
There is no proxy method for ``has`` on the Cache object: https://github.com/sh4nks/flask-caching/blob/master/flask_caching/__init__.py#L261-L297

**Comment by oittaa** [comment]
Thanks, so only proxy methods are exposed from the `Cache` object.
Is there a "proper" way to use a cache object like `SimpleCache` or `NullCache` with an exposed `has` method, because I didn't really find an answer from the documentation?
```
from flask_caching.backends.simplecache import SimpleCache
cache = SimpleCache()
cache.set("a", 123)
print(cache.has("a"))
```

**Comment by sh4nks** [comment]
You can access the backend from the cache object like this:
``cache.cache.has()`` where the first ``cache`` is the Cache object and the second the caching backend.

The reason this is not exposed on the cache object is because it is an optional method. So make sure your caching backend supports ``has``. See https://github.com/sh4nks/flask-caching/blob/master/flask_caching/backends/base.py#L154-L167
